cmd/server/handler/buyer_test.go

```diff
@@ -1 +1,291 @@
+
+	})
+
+	t.Run("Should return err if id is invalid", func(t *testing.T) {
+		server, mockService, handler := InitServerWithGetBuyers(t)
+
+		mockService.On("Get", mock.Anything, "invalid").Return(domain.Buyer{}, buyer.ErrInvalidID)
+
+		server.GET(Get, handler.Get())
+
+		request, response := testutil.MakeRequest(http.MethodGet, "/buyers/invalid", "")
+
+		server.ServeHTTP(response, request)
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+	})
+
+	t.Run("Find by ID status 404 with buyer not found", func(t *testing.T) {
+		server, mockService, handler := InitServerWithGetBuyers(t)
+
+		server.GET(Get, handler.Get())
+
+		request, response := testutil.MakeRequest(http.MethodGet, "/buyers/1", "")
+
+		mockService.On("Get", mock.Anything, 1).Return(domain.Buyer{}, buyer.ErrNotFound)
+
+		server.ServeHTTP(response, request)
+
+		assert.Equal(t, http.StatusNotFound, response.Code)
+
+	})
+}
+
+func TestDelete(t *testing.T) {
+	t.Run("Delete status 204 successful with no content", func(t *testing.T) {
+		server, mockService, handler := InitServerWithGetBuyers(t)
+
+		mockService.On("Delete", mock.Anything, 7).Return(nil)
+
+		request, response := testutil.MakeRequest(http.MethodDelete, "/buyers/7", "")
+
+		server.DELETE(Delete, handler.Delete())
+
+		server.ServeHTTP(response, request)
+
+		assert.Equal(t, http.StatusNoContent, response.Code)
+
+	})
+
+	t.Run("Should return status 404 when buyer is not found", func(t *testing.T) {
+		server, mockService, handler := InitServerWithGetBuyers(t)
+
+		request, response := testutil.MakeRequest(http.MethodDelete, "/buyers/1", "")
+
+		mockService.On("Delete", mock.Anything, 1).Return(buyer.ErrNotFound)
+
+		server.DELETE(Delete, handler.Delete())
+
+		server.ServeHTTP(response, request)
+
+		assert.Equal(t, http.StatusNotFound, response.Code)
+	})
+}
+
+func TestCreate(t *testing.T) {
+	t.Run("Should return status 201 with the buyer created", func(t *testing.T) {
+		server, mockService, handler := InitServerWithGetBuyers(t)
+		expectedBuyers := domain.Buyer{
+			ID:           9,
+			CardNumberID: "2556",
+			FirstName:    "Giulianna",
+			LastName:     "Oliveira",
+		}
+
+		mockService.On("Create", mock.Anything, mock.Anything).Return(expectedBuyers, nil)
+
+		server.POST(Create, handler.Create())
+
+		request, response := testutil.MakeRequest(http.MethodPost, Create, `{
+			"card_number_id":"2556",
+			"first_name":"Giulianna",
+			"last_name":"Oliveira"}`)
+		server.ServeHTTP(response, request)
+
+		responseResult := domain.BuyerResponseID{}
+
+		_ = json.Unmarshal(response.Body.Bytes(), &responseResult)
+		assert.Equal(t, http.StatusCreated, response.Code)
+
+		assert.Equal(t, expectedBuyers, responseResult.Data)
+	})
+
+	t.Run("Should return status 422 when JSON is invalid", func(t *testing.T) {
+		server, _, handler := InitServerWithGetBuyers(t)
+
+		server.POST(Create, handler.Create())
+
+		request, response := testutil.MakeRequest(http.MethodPost, Create, `{"card_number_id":""}`)
```

Comment:
Devolver erro 400 quando for colocado algo vazio na requisição. 422 apenas para recebimento do Bind do JSON